CAN module does not build without the `embassy` feature

After the recently merged [CAN](https://github.com/ch32-rs/ch32-hal/pull/75) support, the builds without the `embassy` feature are broken, as reported by the weekly build of [ch32-hal-template](https://github.com/ch32-rs/ch32-hal-template), see the error [here](https://github.com/ch32-rs/ch32-hal-template/actions/runs/18459828059/job/52601908735#step:8:142):

```
error[E0609]: no field `timeout` on type `can::can::Config`
   --> /home/runner/.cargo/git/checkouts/ch32-hal-f8ddd79b527ebc9f/aded323/src/can/can.rs:220:29
    |
220 |             timeout: config.timeout,
    |                             ^^^^^^^ unknown field
```

I think we have several solutions here:
1. Add support for non-embassy build in `can.rs`
2. Make `embassy` mandatory in `ch32-hal` but it might still be valuable on micro-controllers with small FLASH.
3. Only support the `Can` module with the embassy feature

@mickeprag and @paval-shlyk do you think that solution **1.** is doable without too much rework?

Edit: I have not look at the code yet, the fix might be trivial.

To reproduce the build error:

```
cargo +nightly build --target=riscv32imc-unknown-none-elf --features=rt,ch32l103c8t6 --no-default-features
```

--- a/src/can/can.rs
+++ b/src/can/can.rs
@@ -217,6 +217,7 @@ impl<'d, T: Instance, M: Mode> Can<'d, T, M> {
             _peri: peri,
             fifo,
             last_mailbox_used: usize::MAX,
+            #[cfg(feature = "embassy")]
             timeout: config.timeout,
             _phantom: PhantomData,
         };

